packages/core/lib/commands/test/determineTestFilesToRun.js

```diff
@@ -1,27 +1,65 @@
+const path = require("path");
+const fs = require("fs");
+const glob = require("glob");
+
 const determineTestFilesToRun = ({ inputFile, inputArgs = [], config }) => {
-  const path = require("path");
-  const fs = require("fs");
-  const glob = require("glob");
   let filesToRun = [];
-
   if (inputFile) {
     filesToRun.push(inputFile);
-  } else if (inputArgs.length > 0) {
-    inputArgs.forEach(inputArg => filesToRun.push(inputArg));
+  }else if (inputArgs.length > 0) {
+    //Looping through every argument
+    for(const fileOrDir of inputArgs){
+      let isFileExists = (fs.existsSync(fileOrDir) && fs.statSync(fileOrDir).isFile());
+      let isDirectoryExists = (fs.existsSync(fileOrDir) && fs.statSync(fileOrDir).isDirectory());
+      // If the argument is a file, add the file to filesToRun
+      if(isFileExists){
+        filesToRun.push(fileOrDir);
+      }
+      // if the argument is a directory, run the recursive algorithm to detect every file within the directory and its' subdirectories.
+      else if(isDirectoryExists){
+        const files = getFilesRecursive(fileOrDir,[]);
+        files.forEach((file) => {
+          filesToRun.push(file);
+        });
+      }
+      else{
+
+      }
+    }
   }
 
   if (filesToRun.length === 0) {
     const directoryContents = glob.sync(
-      `${config.test_directory}${path.sep}**${path.sep}*`
+        `${config.test_directory}${path.sep}**${path.sep}*`
     );
     filesToRun =
-      directoryContents.filter(item => fs.statSync(item).isFile()) || [];
+        directoryContents.filter(item => fs.statSync(item).isFile()) || [];
   }
   return filesToRun.filter(file => {
     return file.match(config.test_file_extension_regexp) !== null;
   });
 };
 
+/*
+ * @param {String} dirPath - A string that represents the relative or absolute path of the directory to search through.
+ * @returns {array[String]} - An array of all files in the directory dirPath, including all files of all subdirectories in dirPath.
+ */
+const getFilesRecursive = (dirPath, arrayOfFiles) => {
```

Comment:
This function doesn't consider sym-linked folders. How would this function behave with the following structure below, where `a-prime` is a symbolic link to `a`?
```
a
├── a-prime -> ../a
└── a.txt
```